platform-sdk/docs/core/dynamic-address-book/DynamicAddressBook-PlatformDesign.md

```diff
@@ -0,0 +1,107 @@
+# Dynamic Address Book - Platform Design
+
+![Dynamic Address Book Platform Design](DynamicAddressBook-PlatformDesign.drawio.svg)
+
+## Network Connection Effects
+
+Assumption: That the network is fully connected, all nodes connect to each other.
+
+* Need to validate that this is true.
+* Huge gap in testing if not.
+
+### Add Node
+
+Existing nodes need to create or receive mutual TLS gossip connections with the added node.
+
+This change takes effect when a node is added in the `currentEffectiveRoster` for the `pendingConsensusRound`.
+
+### Remove Node
+
+Existing nodes need to stop gossiping with the removed node. Events from the removed node can still be gossiped,
+validated and come to consensus until the round this new roster takes effect becomes ancient.
+
+This change takes effect when a node is removed from the `currentEffectiveRoster` for the `pendingConsensusRound`.
+
+### Rotate Signing Key
+
+All nodes need to tear down their existing mutual TLS connection with the node that has had its key rotated and
+establish a new connection with the node using the new signing certificate as the trusted root certificate.
+
+This change takes effect when a node's key changes in the `currentEffectiveRoster` for the `pendingConsensusRound`.
+
+### Node Weight Change
+
+This has no effect on the existence of network connections.
+
+## Event Creation Effects
+
+### Add Node
+
+Begin using events from the node as parents of new events.
+
+This change takes effect when a node is added to the `currentEffectiveRoster` for the `pendingConsensusRound`.
+
+This change will modify consensus weight for the network and the event creation algorithm will be reset.
+
+Need to buffer the latest events in each round. The event creator can build on top of any event that has a
+birth round that is less than or equal to the event creator's current round. Events that have a birth round higher
```

Comment:
>The event creator can build on top of any event that has a birth round that is less than or equal to the event creator's current round.

Do you mean the event creator's pending round?

Reply:
yes, I've reorganized the wording, please re-review.